packages/blocks/src/api/validation/index.js

```diff
@@ -407,20 +407,31 @@ export function isEqualTagAttributePairs( actual, expected, logger = createLogge
 		return false;
 	}
 
-	// Convert tuples to object for ease of lookup
-	const [ actualAttributes, expectedAttributes ] = [ actual, expected ].map( fromPairs );
+	// Convert tuples to object for ease of lookup. Using a `for` loop instead of `Array#map` since it is better
+	// optimized. Expected attributes names are normalized to lower case for ease of case-insensitive comparisons later.
+	const actualAttributes = {};
+	const expectedAttributes = {};
+	for ( let i = 0; i < actual.length; i++ ) {
+		actualAttributes[ actual[ i ][ 0 ] ] = actual[ i ][ 1 ];
+		expectedAttributes[ expected[ i ][ 0 ].toLowerCase() ] = expected[ i ][ 1 ];
+	}
 
 	for ( const name in actualAttributes ) {
```

Comment:
>I don't want to let this get carried away, but with the revised implementation, it seems like we could easily combine these two loops into a single loop, since they're both effectively iterating over the same set of actual attributes.

On further review, this isn't possible, because the reason these objects are created separate from the subsequent loop is to support the case that attributes can occur in different orders. However, there's a middle-ground option here, in that only the "expected" comparison needs to be converted to an object. We can avoid creating an `actualAttributes` object, since we'll loop over the actual set anyways in the next loop.

See 57effe969 .

There's probably not much gained through this, but it's a decent simplification to avoid an unnecessary object.

For me, it's enough to offset a "cost" of doing the lower-casing in all code paths. As noted below, I have some concerns where trying to use it as a fallback could leave potential for unexpected behavior.